js/modules/k6/experimental/tracing/trace_id.go

```diff
@@ -61,43 +102,22 @@ type TraceID struct {
 // 4. The remaining bytes are filled with random bytes.
 //
 // The resulting 16 bytes sequence is then encoded as a hex string.
-func (t TraceID) Encode() (string, error) {
+func (t traceID) Encode() (string, error) {
 	if !t.isValid() {
 		return "", fmt.Errorf("failed to encode traceID: %v", t)
 	}
```

Comment:
Should we consider it _generation_?

Reply:
I'm sorry but I don't understand what you mean, could you elaborate? Do you mean we should consider renaming the method, or the error message?

Reply:
The essence of the API is decoupling the _generation_ from _encoding_. I meant that maybe the validation should be part of the generation so it should stay in the constructor. TBH, not sure so mostly asking for your vision on it.

Reply:
Thanks for the clarification 🙇‍♂️That's a fair point, having moved most of the logic in the constructor, the validation should probably happen there indeed considering the `Encode` only runs hexadecimal encoding at this point 🤝